test/integration/test-video-players-helper.js

```diff
@@ -111,6 +117,168 @@ export function runVideoPlayerIntegrationTests(
     afterEach(cleanUp);
   });
 
+  describe.configure().retryOnSaucelabs().run('Video Docking', function() {
+    this.timeout(TIMEOUT);
+
+    describe('General Behavior', () => {
+      it('should have class when attribute is set (autoplay)', function() {
+        return getVideoPlayer(
+            {
+              outsideView: false,
+              autoplay: true,
+              dock: true,
+            }
+        ).then(r => {
+          return poll('checking class list', () => {
+            return !!r.video.classList.contains('i-amphtml-dockable-video');
+          }, undefined, TIMEOUT);
+        });
+      });
+
+      it('should have class when attribute is set (no-autoplay)', function() {
+        return getVideoPlayer(
+            {
+              outsideView: false,
+              autoplay: false,
+              dock: true,
+            }
+        ).then(r => {
+          return poll('checking class list', () => {
+            return !!r.video.classList.contains('i-amphtml-dockable-video');
+          }, undefined, TIMEOUT);
+        });
+      });
+    });
+
+    describe('without-autoplay', () => {
+      it('should minimize when out of viewport', function() {
+        let viewport;
+        let video;
+        return getVideoPlayer(
+            {
+              outsideView: true,
+              autoplay: false,
+              dock: true,
+            }
+        ).then(r => {
+          video = r.video;
+          const playButton = createButton(r, 'play');
+          playButton.click();
+          return listenOncePromise(video, VideoEvents.PLAY);
+        }).then(() => {
+          viewport = video.implementation_.getViewport();
+          // scroll to the bottom, make video fully visible
+          viewport.scrollIntoView(video);
+          return poll('wait for scroll', () => {
+            return video.querySelector('video, iframe')
+             && viewport.getScrollTop() != 0;
+          }, undefined, TIMEOUT);
+        }).then(() => {
+          viewport.setScrollTop(0);
+          return poll('waiting for scroll', () => {
+            return viewport.getScrollTop() == 0;
+          }, undefined, TIMEOUT);
+        }).then(() => {
+          return poll('checking class list', () => {
+            const insideElement = video.querySelector('video, iframe');
+            const classes = insideElement.classList;
+            return !!classes.contains('i-amphtml-dockable-video-minimizing');
+          }, undefined, TIMEOUT);
+        });
```

Comment:
can we add another `then` and assert a bit more? (like presence of `transform: scale()` with the expected scale factor)